src/Our.Umbraco.AuthU/Component/MigrationsRunnerComponent.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Migrations;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Migrations.Upgrade;
+using Umbraco.Core.Scoping;
+using Umbraco.Core.Services;
+using Our.Umbraco.AuthU.Data.Migrations.Plans;
+
+namespace Our.Umbraco.AuthU.Component
+{
+    /// <summary>
+    /// This creates a component we can register as umbraco 8 starts up
+    /// This is done in MigrationRunnerComposer.cs
+    /// </summary>
+    internal class MigrationsRunnerComponent : IComponent
+    {
+        private readonly IScopeProvider _scopeProvider;
+        private readonly IMigrationBuilder _migrationBuilder;
+        private readonly IKeyValueService _keyValueService;
+        private readonly ILogger _logger;
+
+        public MigrationsRunnerComponent(IScopeProvider scopeProvider, IMigrationBuilder migrationBuilder, IKeyValueService keyValueService, ILogger logger)
+        {
+            _scopeProvider = scopeProvider;
+            _migrationBuilder = migrationBuilder;
+            _keyValueService = keyValueService;
+            _logger = logger;
+        }
+
+        public void Initialize()
+        {
+            try
+            {
+                //Create the oAuth tables
+                var upgrader = new Upgrader(new RegisterOAuthTables());
+
+                //Run this plan to remove the oAuth tables, there may be a better way to do this?
+                //var upgrader = new Upgrader(new RemoveOAuthTables());
```

Comment:
To be honest, I don't think we need to worry about downgrading. Umbraco have removed the ability to undo a migration, so I'd say lets just focus on installing.